data/src/main/java/com/linkedin/data/schema/IndentedPdlBuilder.java

```diff
@@ -124,6 +120,16 @@ boolean writeDoc(String doc) throws IOException
     return false;
   }
 
+  @Override
+  PdlBuilder writeJson(Object value) throws IOException
```

Comment:
I don't think this method is necessary. You can avoid adding this logic as an overridden method which calls the super method (a little confusing) by just updating the codec's pretty printer on indent change (i.e. `#increaseIndent` and `#decreaseIndent`). It seems more intuitive IMO to compute the indent string and update the pretty printer on indent change rather than on each use.

Reply:
It shouldn't be confusing at Indenter is kind of decorating the vanilla builder so it should be good to have child class calling super method.

Regarding having this logic in increase and decrease indent I would avoid it as to why build it when it's not needed.